testfixtures/tests/test_compare.py

```diff
@@ -29,23 +32,41 @@ def hexsub(raw):
 
 class TestCompare(TestCase):
 
-    def checkRaises(self, x, y, message=None, regex=None, **kw):
+    def checkRaises(self, x=not_there, y=not_there, message=None, regex=None,
+                    expected=not_there, actual=not_there,
+                    **kw):
         try:
-            compare(x, y, **kw)
+            compare(x, y, expected=expected, actual=actual, **kw)
         except Exception as e:
             if not isinstance(e, AssertionError):
                 self.fail('Expected AssertionError, got %r' % e)
             actual = hexsub(e.args[0])
             if message is not None:
                 # handy for debugging, but can't be relied on for tests!
-                #compare(actual, message, show_whitespace=True)
+                #compare(actual, expected=message, show_whitespace=True)
                 self.assertEqual(actual, message)
             else:
                 if not regex.match(actual): # pragma: no cover
                     self.fail('%r did not match %r' % (actual, regex.pattern))
         else:
             self.fail('No exception raised!')
-            
+
+    def test_insufficient_arguments_raises(self):
+        # More thorough tests in TestGetExpectedActual below, but throwing
+        # in a couple just to make sure things work as expected for compare()
+        msg = ('Exactly 2 of the arguments (x, y, expected, actual) '
+               'must be specified.')
+        with ShouldRaise(ValueError(msg)):
+            compare()
+        with ShouldRaise(ValueError(msg)):
+            compare(None)
+        with ShouldRaise(ValueError(msg)):
+            compare(y=None)
+        with ShouldRaise(ValueError(msg)):
+            compare(expected=None)
+        with ShouldRaise(ValueError(msg)):
+            compare(actual=None)
```

Comment:
These should be separate tests, but hopefully with a simpler approach, less will be needed.